uncertainty_engine/resource_provider.py

```diff
@@ -0,0 +1,364 @@
+import os
+from typing import Any, Optional
+from uuid import uuid4
+
+import requests
+from uncertainty_engine_resource_client.api import ProjectRecordsApi, ResourcesApi
+from uncertainty_engine_resource_client.api_client import ApiClient
+from uncertainty_engine_resource_client.configuration import Configuration
+from uncertainty_engine_resource_client.exceptions import ApiException
+from uncertainty_engine_resource_client.models import (
+    PostResourceRecordRequest,
+    PostResourceVersionRequest,
+    ResourceRecordInput,
+    ResourceVersionRecordInput,
+)
+
+from uncertainty_engine.auth_provider import AuthProvider
+from uncertainty_engine.utils import format_api_error
+
+DATETIME_STRING_FORMAT = "%H:%M:%S %Y-%m-%d"
+DEFAULT_RESOURCE_DEPLOYMENT = "http://localhost:8001/api"
+
+
+class ResourceProvider:
+    """
+    Client for managing resources in the Uncertainty Engine platform.
+
+    This client makes it easy to upload, download, update, and list resources
+    stored in your Uncertainty Engine projects. Use this client when you need to:
+
+    - Upload resources to your projects for later use
+    - Download resources that you or others have previously upload
+    - Update existing resources
+    - Get a list of available resources in a project
+
+    Before using this client, you'll need a project ID and appropriate access rights.
+    """
+
+    def __init__(
+        self,
+        deployment: str = DEFAULT_RESOURCE_DEPLOYMENT,
+        auth_provider: Optional[AuthProvider] = None,
+    ):
+        """
+        Create an instance of a ResourceProvider
+
+        Args:
+            deployment: The URL of the resource service. You typically won't need
+                        to change this unless instructed by support.
+            auth_provider: Handles your authentication. This is usually provided
+                           by the parent application.
+        """
+        self.auth_provider = auth_provider
+        config = Configuration(host=deployment)
+        self.client = ApiClient(configuration=config)
+        self.projects_client = ProjectRecordsApi(self.client)
+        self.resources_client = ResourcesApi(self.client)
+
+    @property
+    def account_id(self) -> Optional[str]:
+        """Get the current account ID from the auth provider"""
```

Comment:
We should add a description of the return here:
```suggestion
        """
        Get the current account ID from the auth provider
        
        Returns:
            If an auth provider is present returns the account ID being used for authentication. Otherwise returns ``None``.             
        """
```